File: src/main/java/backend/model/user/Address.java
package backend.model.user;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Address {

    @Id
    @GeneratedValue
    private Integer id;
    private String street;
    private String housenumber;
    private String city;
    private String country;

    public Address(String street, String housenumber, String city, String country) {
        this.street = street;
        this.housenumber = housenumber;
        this.city = city;
        this.country = country;
    }

    public Address() {

    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getHousenumber() {
        return housenumber;
    }

    public void setHousenumber(String housenumber) {
        this.housenumber = housenumber;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }
}

File: src/main/java/backend/model/user/Address.kt
package backend.model.user

import javax.persistence.Embeddable

@Embeddable
class Address {

    /**
     * Private constructor for JPA
     */
    private constructor() : super()

    constructor(street: String, housenumber: String, city: String, country: String, zipcode: String) : this() {
        this.street = street
        this.housenumber = housenumber
        this.city = city
        this.country = country
        this.zipcode = zipcode
    }

    lateinit var street: String
    lateinit var housenumber: String
    lateinit var city: String
    lateinit var country: String
    lateinit var zipcode: String

}
